NREL/ssc
# SAM 658 restore pv priority manual dispatch

Solves https://github.com/NREL/SAM/issues/658 for manual dispatch.

Note - includes code from https://github.com/NREL/ssc/pull/899 to avoid merge conflicts. Might be best to do code review here after that one is merged.

Pairs with SAM pull request https://github.com/NREL/SAM/pull/1205

To test:

1. Check out this branch and associated SAM branch and build
2. Open a behind the meter PV-battery case
3. Choose manual dispatch, uncheck "charge system only when system power exceeds load", change to "charge battery" priority
4. Run system
5. See that the battery is charging before PV goes to load

Also works with generic system and fuel cell-commercial.
